docs/source/building/variants.rst

```diff
@@ -0,0 +1,1201 @@
+been many proposals (`Conda-build issue
+1142 <https://github.com/conda/conda-build/issues/1142>`_).
+
+As of conda-build 3.0, a new configuration scheme has been added, dubbed
+"variants." Conceptually, this decouples pinning values from recipes, replacing
+them with Jinja2 template variables. It adds support for the notion of
+"compatible" pinnings, though that concept is currently still under heavy
+development, to be integrated with ABI compatibility databases, such as `ABI
+Laboratory <https://abi-laboratory.pro/>`_.
+
+Variant input is ultimately a dictionary. They're mostly very flat. Keys are
+made directly available in Jinja2 templates. As a result, keys in the dictionary
+(and in files read into dictionaries) must be valid jinja2 variable names (no ``-``
+characters allowed). This example builds python 2.7 and 3.5 packages in one
+build command:
+
+conda_build_config.yaml like:
+
+.. code-block:: yaml
+
+   python:
+       - 2.7
+       - 3.5
+
+
+meta.yaml contents like:
+
+.. code-block:: yaml
+
+   package:
+       name: compiled-code
+       version: 1.0
+
+   requirements:
+       build:
+           - python {{ python }}
+       run:
+           - python
+
+
+General pinning examples
+------------------------
+
+There are a few characteristic use cases for pinning.  Please consider this a map for the content below.
+
+1. Shared library providing a binary interface. All uses of this library use the
+   binary interface.  It is convenient to apply the same pin to all of your builds.
+   Example: boost
+
+
+conda_build_config.yaml in your HOME folder:
+
+.. code-block:: yaml
+
+  boost:
+    - 1.61
+    - 1.63
+  pin_run_as_build:
+    boost: x.x
+
+
+meta.yaml:
+
+.. code-block:: yaml
+
+   package:
+       name: compiled-code
+       version: 1.0
+
+   requirements:
+       build:
+           - boost  {{ boost }}
+       run:
+           - boost
+
+This example demonstrates several features:
+  * site-wide configuration with a specifically named config file
+    (conda_build_config.yaml in your home folder). More options below in
+    `Creating conda-build variant config files`_.
+  * building against multiple versions of a single library (set versions
+    installed at build time)
+  * pinning runtime requirements to the version used at build time. More
+    information below at `Pinning at the variant level`_.
+  * specify granularity of pinning. ``x.x`` pins major and minor version. More
+    information at `Pinning expressions`_.
+
+
+2. Python package with externally accessible binary component. Not all uses of
+   this library use the binary interface (some only use pure python). Example:
+   numpy
+
+conda_build_config.yaml in your recipe folder (alongside meta.yaml:
+
+.. code-block:: yaml
+
+  numpy:
+    - 1.11
+    - 1.12
+
+
+meta.yaml:
+
+.. code-block:: yaml
+
+   package:
+       name: numpy_using_thing
+       version: 1.0
+
+   requirements:
+       build:
+           - python
+           - numpy
```

Comment:
Nevermind, after reading `Avoiding unncessary builds` again, I think I understand what you're trying to convey.  I'm going to submit a PR to your branch to help clarify this example.